urwid/widget.py

```diff
@@ -1412,7 +1414,7 @@ def _normalize_to_caption(self, text):
             return text
         if tu:
             return text.encode('ascii') # follow python2's implicit conversion
-        return text.decode('ascii')
+        return text.decode(sys.stdin.encoding)
```

Comment:
No.

As I said, the screen is the only place that has a chance of knowing the right encoding.

If you want the current version urwid to not crash on whatever encoding the user has selected you can use a bytestring for the caption in the Edit widget and the bytes will be passed through without attempting to encode or decode them. (admittedly this guesses the type of the encoding: narrow/wide based on the local console for historical reasons)

If you want to decode the correct encoding the only place that this can be done is the screen. Widgets have no idea where the user input came from and can't assume they came from sys.stdin. A future version of urwid should probably just use Unicode everywhere by having the screen decode and encode all characters. The reason I didn't do this originally was because some encodings don't round-trip to Unicode and back safely. That might not be a real concern these days, I'm not sure.